react/finger-exercises/ecmascript/02 - promises/index.js

```diff
@@ -1,9 +1,20 @@
 // Hint: use setInterval, create a new Promise and measure time with Date.now()
 
-export function delay() {
+export function delay(time) {
+  const startedTime = Date.now();
+  return new Promise((resolve, reject) => setInterval(() => {
+    if (time >= 5000) {
+      reject(Error('This time is too much !'));
+    }
+    const endTime = Date.now();
+    const elapsedTime = endTime - startedTime;
 
+    if (elapsedTime >= time) {
+      resolve(elapsedTime);
+    }
+  }, 100));
 }
 
-export function asyncDelay() {
-
+export async function asyncDelay(time) {
```

Comment:
There's no need to make this an `async` function.

Reply:
Totalmente,  corregido